flang/lib/Lower/IO.cpp

```diff
@@ -249,6 +249,11 @@ genOutputItemList(Fortran::lower::AbstractConverter &converter,
           itemValue);
       outputFuncArgs.push_back(parts.first);
       outputFuncArgs.push_back(parts.second);
+    } else if (fir::isa_ref_type(itemType)) {
```

Comment:
I was expecting a test for the forms of `CHARACTER` here. Can't we have a `!fir.ref<!fir.array<10x!fir.char<1>>>` type for a variable of `CHARACTER(LEN=10)`?

Reply:
I am not entirely sure why but arrays like:
```
INTEGER :: A(10)
LOGICAL :: B(10)
REAL :: C(10)
```
reach and are caught by that branch, but character arrays like:
```
CHARACTER :: D(10)
```
lower without any errors.